### DISCUSSION TITLE
Cargo workspace external types

### DISCUSSION THREAD

**Discussion by vhdirk** [fzyzcjy/flutter_rust_bridge#1911] [Q&A] [answered]
So I'm developing a rust library that I want to create dart/flutter bindings for. My project is a cargo workspace that currently has this structure:

lox: project root
 - lox: main library, all other crates in the workspace depend on this.
 - lox-cli: cli tools
 - lox-py: build cython extension. Structs are annotated in the main library, gated with a feature flag. This crate just builds the dylib.
 - lox-dart: here's where the bindings for dart should go

Given that a crate can be multiple bins but just one lib, I don't see how I could structure my project in any other way.

However, flutter_rust_bridge only works for structs in the same crate (lox-dart), I would have to mirror every single type of my library. Which seems all but practical or efficient.
What would be the best approach here? Is there an annotation (like pyo3 has) for struct in the structs/methods in the main library?

Thanks!

**Comment by fzyzcjy** [comment, marked as answer]
Hi, does the proposal in https://github.com/fzyzcjy/flutter_rust_bridge/issues/1903 solves your problem? (Feel free to discuss there as well)

**Comment by acul009** [comment]
I have my flutter bindings and api in a subfolder (crate in workspace) called flutter_bridge/

You can still use your libraries, but you'll have to treat them as external libraries:
https://cjycode.com/flutter_rust_bridge/guides/types/translatable/external/diff-crate

Check the linked issue for examples

**Reply by vhdirk** [reply]
Thanks, I'll check it out. It's not ideal, but it'll have to do :)

**Comment by fzyzcjy** [comment]
Now automatically translating a whole third party crate is possible: https://cjycode.com/flutter_rust_bridge/guides/third-party
